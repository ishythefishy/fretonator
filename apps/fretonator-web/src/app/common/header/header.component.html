<header class="header">
  <a
    [routerLink]="['/']"
    class="header__siteLogo"
    aria-label="Navigate to home"
  >
    <app-logo-with-text></app-logo-with-text>
  </a>

  <button class="hamburger"
          [class.is__open]="isMenuVisible"
          (click)="toggleMenu()"
          [attr.aria-checked]="isMenuVisible"
          aria-label="Menu Toggle"
          aria-controls="headerLinks"
          role="switch"
          type="button">
    <span class="hamburger__box">
      <span class="hamburger__inner"></span>
    </span>
  </button>

  <ul class="header__links"
      id="headerLinks"
      [attr.aria-expanded]="isMenuVisible"
      [class.header__links--hide]="!isMenuVisible">
    <li class="header__linksItem">
      <a [routerLink]="['/']"
         [routerLinkActiveOptions]="{exact:true}"
         routerLinkActive="header__linksItemLink--active"
         class="header__linksItemLink"
      >Home</a
      >
    </li>
    <li class="header__linksItem">
      <a [routerLink]="['/', 'about']"
         routerLinkActive="header__linksItemLink--active"
         class="header__linksItemLink"
      >About</a
      >
    </li>
  </ul>
</header>
