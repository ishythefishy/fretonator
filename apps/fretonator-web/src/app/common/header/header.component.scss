@import "../../../styles/vars";
@import "../../../styles/functions";
@import "../../../styles/mixins";

.header {
  background-color: var(--black);
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  padding-top: pxToRem($grid-unit);
}

.header__siteLogo {
  display: flex;
  justify-content: center;
  align-items: center;
  height: pxToRem(120);
  width: pxToRem(120);
  cursor: pointer;
  margin-left: auto;
  margin-right: auto;

  @include focus_accessible();

  @media screen and (min-width: $screen-med) {
    height: pxToRem(140);
    width: pxToRem(140);
  }
}

.hamburger {
  padding: $hamburger_padding-y $hamburger_padding-x;
  display: flex;
  justify-content: flex-start;
  cursor: pointer;

  transition-property: opacity, filter;
  transition-duration: 0.15s;
  transition-timing-function: linear;

  font: inherit;
  color: inherit;
  text-transform: none;
  background-color: transparent;
  border: 0;
  margin: 0;
  overflow: visible;

  &:hover {
    opacity: $hamburger_hover-opacity;
  }

  &.is__open {
    &:hover {
      opacity: $hamburger_active-hover-opacity;
    }
  }

  @include focus_accessible($focus_box_shadow_color);

  @media screen and (min-width: $screen-med) {
    display: none;
  }
}

.hamburger__box {
  width: $hamburger_layer-width;
  height: ($hamburger_layer-height * 3) + ($hamburger_layer-spacing * 2);
  display: inline-block;
  position: relative;
}

.hamburger__inner {
  display: block;
  top: $hamburger_layer-height / 2;
  margin-top: $hamburger_layer-height / -2;
  background-color: $hamburger_active-layer-color;
  transition: background-color 0s 0.13s linear;

  &,
  &:before,
  &:after {
    content: '';
    display: block;
    width: $hamburger_layer-width;
    height: $hamburger_layer-height;
    background-color: $hamburger_layer-color;
    border-radius: $hamburger_layer-border-radius;
    position: absolute;
    transition-property: transform;
    transition-duration: 0.15s;
    transition-timing-function: ease;
  }

  &:before {
    top: $hamburger_layer-height + $hamburger_layer-spacing;
    transition: top 0.1s 0.2s cubic-bezier(0.33333, 0.66667, 0.66667, 1),
    transform 0.13s cubic-bezier(0.55, 0.055, 0.675, 0.19);
  }

  &:after {
    top: ($hamburger_layer-height * 2) + ($hamburger_layer-spacing * 2);
    transition: top 0.2s 0.2s cubic-bezier(0.33333, 0.66667, 0.66667, 1),
    transform 0.13s cubic-bezier(0.55, 0.055, 0.675, 0.19);
  }
}

.is__open {
  .hamburger__inner {
    transition-delay: 0.22s;
    background-color: transparent !important;

    &:before {
      top: 0;
      transition: top 0.1s 0.15s cubic-bezier(0.33333, 0, 0.66667, 0.33333),
      transform 0.13s 0.22s cubic-bezier(0.215, 0.61, 0.355, 1);
      transform: translate3d(0, $hamburger_layer-spacing + $hamburger_layer-height, 0) rotate(45deg);
    }

    &:after {
      top: 0;
      transition: top 0.2s cubic-bezier(0.33333, 0, 0.66667, 0.33333),
      transform 0.13s 0.22s cubic-bezier(0.215, 0.61, 0.355, 1);
      transform: translate3d(0, $hamburger_layer-spacing + $hamburger_layer-height, 0) rotate(-45deg);
    }
  }
}

.header__links {
  list-style: none;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  padding: pxToRem($grid-unit) pxToRem($grid-unit) pxToRem($grid-unit * 2) pxToRem($grid-unit);
  text-align: center;
  display: block;
}

.header__links--hide {
  display: none;

  @media screen and (min-width: $screen-med) {
    display: block !important;
  }
}

.header__linksItem {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: pxToRem($grid-unit * 2) pxToRem($grid-unit);

  @media screen and (min-width: $screen-med) {
    display: inline-block;
  }
}

.header__linksItemLink {
  color: var(--peach);
  font-size: pxToRem(14);
  font-weight: var(--font-weight-bold);
  letter-spacing: var(--letter-spacing-display);
  text-decoration: none;
  text-transform: uppercase;
  transition: color 0.2s ease-in-out;

  @include focus_accessible();

  &:hover,
  &:focus {
    color: var(--yellow);
  }
}

.header__linksItemLink--active {
  border-bottom: pxToRem(2) solid var(--peach);
}
