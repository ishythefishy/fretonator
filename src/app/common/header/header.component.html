<header class="header">
  <a routerLink="/" class="header__siteLogo"
    ><img class="header__siteLogoImage" src="/assets/logo.png"
  /></a>
  <ul class="header__links">
    <li href="" class="header__linksItem">
      <a routerLink="/about" class="header__linksItemLink">About</a>
    </li>
    <li href="" class="header__linksItem">
      <a routerLink="/contact" class="header__linksItemLink">Contact</a>
    </li>
  </ul>
</header>
