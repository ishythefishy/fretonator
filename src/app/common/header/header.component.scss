@import '../../../styles/vars';
@import '../../../styles/functions';
@import '../../../styles/mixins';

.header {
  background-color: var(--black);
  height: pxToRem(160);
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
}

.header__siteLogo {
  display: flex;
  flex-direction: column;
}

.header__siteLogoImage {
  height: 100%;
  object-fit: contain;
}

.header__links {
  list-style: none;
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  margin: 0;
  padding-left: 0;
  padding-right: 0;
  padding-top: pxToRem($grid-unit * 2);
  padding-bottom: pxToRem($grid-unit * 2);
  text-align: center;
}

.header__linksItem {
  display: inline-block;
  padding-left: pxToRem($grid-unit);
  padding-right: pxToRem($grid-unit);
}

.header__linksItemLink {
  color: var(--peach);
  font-size: pxToRem(14);
  letter-spacing: var(--letter-spacing-display);
  text-decoration: none;
  text-transform: uppercase;
  transition: color 0.2s ease-in-out;

  @include focus_accessible(4px);

  &:hover,
  &:focus {
    color: var(--yellow);
  }
}
