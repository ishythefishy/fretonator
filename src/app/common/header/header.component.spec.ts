import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';
import { HeaderComponent } from './header.component';
import { Component, DebugElement } from '@angular/core';
import { HeaderModule } from './header.module';

describe('HeaderComponent', () => {

  const selectors = {
    logo: By.css('.header__siteLogo'),
    links: By.css('.header__linksItemLink'),
  };

  @Component({
    selector: 'app-header-spec',
    template: `
      <app-header></app-header>
    `,
  })
  class HeaderComponentSpec {
  }

  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HeaderComponentSpec],
      imports: [RouterTestingModule, HeaderModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('Logo', () => {
    it('should have a link to the home page', () => {
      const logoElem = fixture.debugElement.query(selectors.logo);
      expect(logoElem.properties.href).toBe('/');
    });
  });

  describe('Links', () => {

    let about: DebugElement;
    let contact: DebugElement;

    beforeEach(() => {
      const links = fixture.debugElement.queryAll(selectors.links);
      about = links[0];
      contact = links[1];
    });

    it('About should link to /about', () => {
      expect(about.properties.href).toBe('/about');
    });

    it('Contact should link to /contact', () => {
      expect(contact.properties.href).toBe('/contact');
    });

  });

});
