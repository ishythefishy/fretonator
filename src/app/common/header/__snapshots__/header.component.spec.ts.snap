// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match snapshot 1`] = `
<app-header>
  <header
    class="header"
  >
    <a
      class="header__siteLogo"
      href="/"
    >
      <img
        alt="Fretonator"
        class="header__siteLogoImage"
        src="/assets/logo.png"
      />
    </a>
    <ul
      class="header__links"
    >
      <li
        class="header__linksItem"
      >
        <a
          class="header__linksItemLink"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="header__linksItem"
      >
        <a
          class="header__linksItemLink"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </header>
</app-header>
`;
